--- a/openFrameworks/ofMain.h
+++ b/openFrameworks/ofMain.h
@@ -77,8 +77,10 @@
 
 //--------------------------
 // video
+#ifndef TARGET_NO_VIDEO
 #include "ofVideoGrabber.h"
 #include "ofVideoPlayer.h"
+#endif
 
 //--------------------------
 // events
--- a/openFrameworks/app/ofAppRunner.cpp
+++ b/openFrameworks/app/ofAppRunner.cpp
@@ -21,7 +21,7 @@
 
 // TODO: closing seems wonky.
 // adding this for vc2010 compile: error C3861: 'closeQuicktime': identifier not found
-#if defined (TARGET_WIN32) || defined(TARGET_OSX)
+#if !defined(TARGET_NO_VIDEO) && (defined(TARGET_WIN32) || defined(TARGET_OSX))
 	#include "ofQtUtils.h"
 #endif
 
@@ -353,7 +353,7 @@
 	#endif
 
 	// try to close quicktime, for non-linux systems:
-	#if defined(OF_VIDEO_CAPTURE_QUICKTIME) || defined(OF_VIDEO_PLAYER_QUICKTIME)
+	#if !defined(TARGET_NO_VIDEO) && (defined(OF_VIDEO_CAPTURE_QUICKTIME) || defined(OF_VIDEO_PLAYER_QUICKTIME))
 	closeQuicktime();
 	#endif
 
--- a/openFrameworks/utils/ofSystemUtils.cpp
+++ b/openFrameworks/utils/ofSystemUtils.cpp
@@ -74,7 +74,9 @@
 
 #if defined( TARGET_LINUX ) && defined (OF_USING_GTK)
 #include <gtk/gtk.h>
+#ifndef TARGET_NO_VIDEO
 #include "ofGstUtils.h"
+#endif
 #include "Poco/Condition.h"
 
 #if GTK_MAJOR_VERSION>=3
@@ -196,7 +198,9 @@
 		#endif
 		int argc=0; char **argv = NULL;
 		gtk_init (&argc, &argv);
+#		ifndef TARGET_NO_VIDEO
 		ofGstUtils::startGstMainLoop();
+#		endif
 		initialized = true;
 	}
 
@@ -208,6 +212,7 @@
 	dialogData.action = action;
 	dialogData.windowTitle = windowTitle;
 	dialogData.defaultName = defaultName;
+#	ifndef TARGET_NO_VIDEO
 	dialogData.done = false;
 
 	g_main_context_invoke(g_main_loop_get_context(ofGstUtils::getGstMainLoop()), &file_dialog_gtk, &dialogData);
@@ -215,6 +220,7 @@
 		dialogData.mutex.lock();
 		dialogData.condition.wait(dialogData.mutex);
 	}
+#	endif
 	return dialogData.results;
 }
 
@@ -282,12 +288,14 @@
 		initGTK();
 		TextDialogData dialogData;
 		dialogData.text = errorMessage;
+#		ifndef TARGET_NO_VIDEO
 		dialogData.done = false;
 		g_main_context_invoke(g_main_loop_get_context(ofGstUtils::getGstMainLoop()), &alert_dialog_gtk, &dialogData);
 		if(!dialogData.done){
 			dialogData.mutex.lock();
 			dialogData.condition.wait(dialogData.mutex);
 		}
+#		endif
 	#endif
 
 	#ifdef TARGET_ANDROID
@@ -595,13 +603,17 @@
 	initGTK();
 	TextDialogData dialogData;
 	dialogData.text = text;
+#	ifndef TARGET_NO_VIDEO
 	dialogData.done = false;
+#	endif
 	dialogData.question = question;
+#	ifndef TARGET_NO_VIDEO
 	g_main_context_invoke(g_main_loop_get_context(ofGstUtils::getGstMainLoop()), &text_dialog_gtk, &dialogData);
 	if(!dialogData.done){
 		dialogData.mutex.lock();
 		dialogData.condition.wait(dialogData.mutex);
 	}
+#	endif
 	text = dialogData.text;
 #endif
 
