--- a/openFrameworks/gl/ofVbo.cpp
+++ b/openFrameworks/gl/ofVbo.cpp
@@ -327,7 +327,7 @@
 	}
 #else
 	if(!vaoChecked){
-		supportVAOs = ofGetGLProgrammableRenderer();// || glewIsSupported("GL_ARB_vertex_array_object"); <- this should work but has false positives on some cards like emulation in vm's
+		supportVAOs = ofIsGLProgrammableRenderer();
 		vaoChecked = true;
 	}
 #endif
--- a/openFrameworks/app/ofAppGLFWWindow.cpp
+++ b/openFrameworks/app/ofAppGLFWWindow.cpp
@@ -15,6 +15,7 @@
 	#else
 		#define GLFW_EXPOSE_NATIVE_EGL
 	#endif
+	#include <X11/extensions/Xrandr.h>
 	#include "GLFW/glfw3native.h"
 	#include <X11/Xatom.h>
 #elif defined(TARGET_OSX)
