--- a/openFrameworks/app/ofAppGLFWWindow.cpp
+++ b/openFrameworks/app/ofAppGLFWWindow.cpp
@@ -942,7 +943,7 @@
 }
 
 //------------------------------------------------------------
-void ofAppGLFWWindow::keyboard_cb(GLFWwindow* windowP_, int keycode, int scancode, unsigned int codepoint, int action, int mods) {
+void ofAppGLFWWindow::keyboard_cb(GLFWwindow* windowP_, int keycode, int scancode, int action, int mods) {
 	int key;
 	switch (keycode) {
 		case GLFW_KEY_ESCAPE:
@@ -1051,14 +1052,13 @@
 			key = OF_KEY_TAB;
 			break;   
 		default:
-			key = codepoint;
 			break;
 	}
 
 	if(action == GLFW_PRESS || action == GLFW_REPEAT){
-		ofNotifyKeyPressed(key,keycode,scancode,codepoint);
+		ofNotifyKeyPressed(key,keycode,scancode);
 	}else if (action == GLFW_RELEASE){
-		ofNotifyKeyReleased(key,keycode,scancode,codepoint);
+		ofNotifyKeyReleased(key,keycode,scancode);
 	}
 }
 
--- a/openFrameworks/app/ofAppGLFWWindow.h
+++ b/openFrameworks/app/ofAppGLFWWindow.h
@@ -115,7 +115,7 @@
 
 	static void 	mouse_cb(GLFWwindow* windowP_, int button, int state, int mods);
 	static void 	motion_cb(GLFWwindow* windowP_, double x, double y);
-	static void 	keyboard_cb(GLFWwindow* windowP_, int key, int scancode, unsigned int codepoint, int action, int mods);
+	static void 	keyboard_cb(GLFWwindow* windowP_, int key, int scancode, int action, int mods);
 	static void 	resize_cb(GLFWwindow* windowP_, int w, int h);
 	static void 	exit_cb(GLFWwindow* windowP_);
 	static void		scroll_cb(GLFWwindow* windowP_, double x, double y);
--- a/openFrameworks/events/ofEvents.cpp
+++ b/openFrameworks/events/ofEvents.cpp
@@ -178,7 +178,7 @@
 }
 
 //------------------------------------------
-void ofNotifyKeyPressed(int key, int keycode, int scancode, int codepoint){
+void ofNotifyKeyPressed(int key, int keycode, int scancode){
 	static ofKeyEventArgs keyEventArgs;
 	// FIXME: modifiers are being reported twice, for generic and for left/right
 	// add operators to the arguments class so it can be checked for both
@@ -208,7 +208,6 @@
 	keyEventArgs.key = key;
 	keyEventArgs.keycode = keycode;
 	keyEventArgs.scancode = scancode;
-	keyEventArgs.codepoint = codepoint;
 	ofNotifyEvent( ofEvents().keyPressed, keyEventArgs );
 	
 	
@@ -220,7 +219,7 @@
 }
 
 //------------------------------------------
-void ofNotifyKeyReleased(int key, int keycode, int scancode, int codepoint){
+void ofNotifyKeyReleased(int key, int keycode, int scancode){
 	static ofKeyEventArgs keyEventArgs;
 
 	// FIXME: modifiers are being reported twice, for generic and for left/right
@@ -251,7 +250,6 @@
 	keyEventArgs.key = key;
 	keyEventArgs.keycode = keycode;
 	keyEventArgs.scancode = scancode;
-	keyEventArgs.codepoint = codepoint;
 	ofNotifyEvent( ofEvents().keyReleased, keyEventArgs );
 }
 
--- a/openFrameworks/events/ofEvents.h
+++ b/openFrameworks/events/ofEvents.h
@@ -44,7 +44,6 @@
 	int key;
 	int keycode;
 	int scancode;
-	unsigned int codepoint;
 };
 
 class ofMouseEventArgs : public ofEventArgs, public ofVec2f {
@@ -252,8 +251,8 @@
 void ofNotifyUpdate();
 void ofNotifyDraw();
 
-void ofNotifyKeyPressed(int key, int keycode=-1, int scancode=-1, int codepoint=-1);
-void ofNotifyKeyReleased(int key, int keycode=-1, int scancode=-1, int codepoint=-1);
+void ofNotifyKeyPressed(int key, int keycode=-1, int scancode=-1);
+void ofNotifyKeyReleased(int key, int keycode=-1, int scancode=-1);
 void ofNotifyKeyEvent(const ofKeyEventArgs & keyEvent);
 
 void ofNotifyMousePressed(int x, int y, int button);
