--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -129,10 +129,7 @@
 endif(CMAKE_SYSTEM MATCHES "SunOS")
 
 if (CMAKE_COMPILER_IS_MINGW)
-    add_definitions(-DWC_NO_BEST_FIT_CHARS=0x400  -DPOCO_WIN32_UTF8)
-    add_definitions(-mno-cygwin -D_WIN32 -DMINGW32 -DWINVER=0x500 -DODBCVER=0x0300 -DPOCO_THREAD_STACK_SIZE -DFoundation_Config_INCLUDED )
-  link_directories(/usr/local/lib /usr/lib)
-  include_directories(/usr/local/include /usr/include)
+  add_definitions(-D_WIN32 -DMINGW32 -DWINVER=0x500 -DODBCVER=0x0300 -DPOCO_THREAD_STACK_SIZE -DFoundation_Config_INCLUDED )
 endif (CMAKE_COMPILER_IS_MINGW)
 
 if (CMAKE_COMPILER_IS_CYGWIN)
@@ -160,9 +160,6 @@
 
 add_subdirectory(Foundation)
 add_subdirectory(XML)
-add_subdirectory(JSON)
-add_subdirectory(MongoDB)
-add_subdirectory(PDF)
 add_subdirectory(Util)
 add_subdirectory(Net)
 #add_subdirectory(Web)
@@ -174,7 +174,6 @@
     add_subdirectory(Crypto)
 endif(OPENSSL_FOUND)
 
-add_subdirectory(Data)
 #add_subdirectory(WebWidgets)
 add_subdirectory(Zip)
 
--- a/Foundation/CMakeLists.txt
+++ b/Foundation/CMakeLists.txt
@@ -15,6 +15,7 @@
       set(kit_bindir "${kit_dir}/bin/x86")
     endif (X64)
   endif ()
+  if ("${CMAKE_GENERATOR}" MATCHES "Visual Studio")
   find_program(CMAKE_MC_COMPILER mc.exe HINTS "${sdk_bindir}" "${kit_bindir}"
     DOC "path to message compiler")
   if (NOT CMAKE_MC_COMPILER)
@@ -22,6 +23,7 @@
   endif (NOT CMAKE_MC_COMPILER)
   message(STATUS "Found message compiler: ${CMAKE_MC_COMPILER}")
   mark_as_advanced(CMAKE_MC_COMPILER)
+  endif ()
 endif(WIN32)
   
 set( BASE_SRCS 
@@ -209,19 +209,17 @@
 )
 
 if (WIN32)
-  set(RES_SRCS ${CMAKE_SOURCE_DIR}/Foundation/src/pocomsg.h)
-  set_source_files_properties(${RES_SRCS} PROPERTIES GENERATED true)
-  add_custom_command(
-    OUTPUT ${RES_SRCS}
-    DEPENDS ${CMAKE_SOURCE_DIR}/Foundation/src/pocomsg.mc
-    COMMAND ${CMAKE_MC_COMPILER}
-    ARGS 
-         -h ${CMAKE_SOURCE_DIR}/Foundation/src
-         -r ${CMAKE_SOURCE_DIR}/Foundation/src
-         ${CMAKE_SOURCE_DIR}/Foundation/src/pocomsg.mc
-    VERBATIM # recommended: p260
+    add_definitions(
+         -DPOCO_CTG_FATAL=0x00000001L
+         -DPOCO_CTG_CRITICAL=0x00000002L
+         -DPOCO_CTG_ERROR=0x00000003L
+         -DPOCO_CTG_WARNING=0x00000004L
+         -DPOCO_CTG_NOTICE=0x00000005L
+         -DPOCO_CTG_INFORMATION=0x00000006L
+         -DPOCO_CTG_DEBUG=0x00000007L
+         -DPOCO_CTG_TRACE=0x00000008L
+         -DPOCO_MSG_LOG=0x00001000L
   )
-  list(APPEND WIN_SRCS ${RES_SRCS})
 endif (WIN32)
 
 set(LIN_SRCS
