--- a/openFrameworks/app/ofAppRunner.cpp
+++ b/openFrameworks/app/ofAppRunner.cpp
@@ -22,9 +22,9 @@
 
 // TODO: closing seems wonky.
 // adding this for vc2010 compile: error C3861: 'closeQuicktime': identifier not found
-#if defined (TARGET_WIN32) || defined(TARGET_OSX)
-	#include "ofQtUtils.h"
-#endif
+//#if defined (TARGET_WIN32) || defined(TARGET_OSX)
+//	#include "ofQtUtils.h"
+//#endif
 
 //========================================================================
 // static variables:
@@ -249,9 +249,9 @@ void ofExitCallback(){
 	#endif
 
 	// try to close quicktime, for non-linux systems:
-	#if defined(OF_VIDEO_CAPTURE_QUICKTIME) || defined(OF_VIDEO_PLAYER_QUICKTIME)
-	closeQuicktime();
-	#endif
+//	#if defined(OF_VIDEO_CAPTURE_QUICKTIME) || defined(OF_VIDEO_PLAYER_QUICKTIME)
+//	closeQuicktime();
+//	#endif
 
 
 	//------------------------
--- a/openFrameworks/utils/ofSystemUtils.cpp
+++ b/openFrameworks/utils/ofSystemUtils.cpp
@@ -74,12 +74,12 @@
 
 #if defined( TARGET_LINUX ) && defined (OF_USING_GTK)
 #include <gtk/gtk.h>
-#include "ofGstUtils.h"
+//#include "ofGstUtils.h"
 
 static void initGTK(){
 	static bool initialized = false;
 	if(!initialized){
-		ofGstUtils::startGstMainLoop();
+//		ofGstUtils::startGstMainLoop();
 	    gdk_threads_init();
 	int argc=0; char **argv = NULL;
 	gtk_init (&argc, &argv);
--- a/openFrameworks/ofMain.h
+++ b/openFrameworks/ofMain.h
@@ -68,13 +68,13 @@
 
 //--------------------------
 // audio
-#include "ofSoundStream.h"
-#include "ofSoundPlayer.h"
+//#include "ofSoundStream.h"
+//#include "ofSoundPlayer.h"
 
 //--------------------------
 // video
-#include "ofVideoGrabber.h"
-#include "ofVideoPlayer.h"
+//#include "ofVideoGrabber.h"
+//#include "ofVideoPlayer.h"
 
 //--------------------------
 // events
