--- a/openFrameworks/utils/ofURLFileLoader.cpp
+++ b/openFrameworks/utils/ofURLFileLoader.cpp
@@ -102,7 +102,7 @@
 void ofURLFileLoader::start() {
      if (isThreadRunning() == false){
 		ofAddListener(ofEvents().update,this,&ofURLFileLoader::update);
-        startThread(true, false);   // blocking, verbose
+        startThread(true);   // blocking, verbose
     }else{
     	ofLogVerbose("ofURLFileLoader") << "start(): signaling new request condition";
     	condition.signal();
