--- a/openFrameworks/app/ofAppGLFWWindow.cpp
+++ b/openFrameworks/app/ofAppGLFWWindow.cpp
@@ -14,6 +14,7 @@
 	#else
 		#define GLFW_EXPOSE_NATIVE_EGL
 	#endif
+	#include <X11/extensions/Xrandr.h>
 	#include "GLFW/glfw3native.h"
 	#include <X11/Xatom.h>
 	#include "Poco/URI.h"
@@ -905,13 +906,14 @@ void ofAppGLFWWindow::scroll_cb(GLFWwindow* windowP_, double x, double y) {
 }
 
 //------------------------------------------------------------
-void ofAppGLFWWindow::drop_cb(GLFWwindow* windowP_, const char* dropString) {
-	string drop = dropString;
+void ofAppGLFWWindow::drop_cb(GLFWwindow* windowP_, const int dropArraySize, const char* dropStringArray[]) {
 	ofDragInfo drag;
 	drag.position.set(ofGetMouseX()*instance->pixelScreenCoordScale, ofGetMouseY()*instance->pixelScreenCoordScale);
-	drag.files = ofSplitString(drop,"\n",true);
+	for(int i=0; i<dropArraySize; i++){
+		drag.files.push_back(dropStringArray[i]);
+	}
 #ifdef TARGET_LINUX
-	for(int i=0; i<(int)drag.files.size(); i++){
+	for(int i=0; i<dropArraySize; i++){
 		drag.files[i] = Poco::URI(drag.files[i]).getPath();
 	}
 #endif
--- a/openFrameworks/app/ofAppGLFWWindow.h
+++ b/openFrameworks/app/ofAppGLFWWindow.h
@@ -113,7 +113,7 @@ private:
 	static void 	resize_cb(GLFWwindow* windowP_, int w, int h);
 	static void 	exit_cb(GLFWwindow* windowP_);
 	static void		scroll_cb(GLFWwindow* windowP_, double x, double y);
-	static void 	drop_cb(GLFWwindow* windowP_, const char* dropString);
+	static void 	drop_cb(GLFWwindow* windowP_, const int dropArraySize, const char* dropStringArray[]);
 	static void		error_cb(int errorCode, const char* errorDescription);
 	static void 	exitApp();
 
