--- a/openFrameworks/app/ofAppGLFWWindow.cpp
+++ b/openFrameworks/app/ofAppGLFWWindow.cpp
@@ -902,7 +902,15 @@ void ofAppGLFWWindow::motion_cb(GLFWwindow* windowP_, double x, double y) {
 
 //------------------------------------------------------------
 void ofAppGLFWWindow::scroll_cb(GLFWwindow* windowP_, double x, double y) {
-	//TODO: implement scroll events
+    int button;
+
+    if (y == 1) {
+        button = OF_MOUSE_BUTTON_4;
+    } else if (y == -1) {
+        button = OF_MOUSE_BUTTON_5;
+    }
+
+    ofNotifyMousePressed(ofGetMouseX(), ofGetMouseY(), button);
 }
 
 //------------------------------------------------------------
