diff --git a/libs/openFrameworks/app/ofAppGLFWWindow.cpp b/libs/openFrameworks/app/ofAppGLFWWindow.cpp
index 02b1a8c..fe7ddc3 100644
--- a/libs/openFrameworks/app/ofAppGLFWWindow.cpp
+++ b/libs/openFrameworks/app/ofAppGLFWWindow.cpp
@@ -14,6 +14,7 @@
 	#else
 		#define GLFW_EXPOSE_NATIVE_EGL
 	#endif
+    #include <X11/extensions/Xrandr.h>
 	#include "GLFW/glfw3native.h"
 	#include <X11/Xatom.h>
 	#include "Poco/URI.h"
diff --git a/libs/openFrameworks/app/ofAppGlutWindow.cpp b/libs/openFrameworks/app/ofAppGlutWindow.cpp
index ab7ec93..ab291fd 100644
--- a/libs/openFrameworks/app/ofAppGlutWindow.cpp
+++ b/libs/openFrameworks/app/ofAppGlutWindow.cpp
@@ -169,10 +169,10 @@ static void fixCloseWindowOnWin32(){
 	DragAcceptFiles (handle, TRUE);
 
 	//store the current message event handler for the window
-	currentWndProc = (WNDPROC)GetWindowLongPtr(handle, GWL_WNDPROC);
+    currentWndProc = (WNDPROC)GetWindowLongPtr(handle, GWLP_WNDPROC);
 
 	//tell the window to now use our event handler!
-	SetWindowLongPtr(handle, GWL_WNDPROC, (long)winProc);
+    SetWindowLongPtr(handle, GWLP_WNDPROC, (long)winProc);
 }
 
 #endif
diff --git a/libs/openFrameworks/app/ofAppRunner.cpp b/libs/openFrameworks/app/ofAppRunner.cpp
index 027bb79..b7f72fb 100644
--- a/libs/openFrameworks/app/ofAppRunner.cpp
+++ b/libs/openFrameworks/app/ofAppRunner.cpp
@@ -22,9 +22,9 @@
 
 // TODO: closing seems wonky.
 // adding this for vc2010 compile: error C3861: 'closeQuicktime': identifier not found
-#if defined (TARGET_WIN32) || defined(TARGET_OSX)
-	#include "ofQtUtils.h"
-#endif
+//#if defined (TARGET_WIN32) || defined(TARGET_OSX)
+//	#include "ofQtUtils.h"
+//#endif
 
 //========================================================================
 // static variables:
@@ -249,9 +249,9 @@ void ofExitCallback(){
 	#endif
 
 	// try to close quicktime, for non-linux systems:
-	#if defined(OF_VIDEO_CAPTURE_QUICKTIME) || defined(OF_VIDEO_PLAYER_QUICKTIME)
-	closeQuicktime();
-	#endif
+//	#if defined(OF_VIDEO_CAPTURE_QUICKTIME) || defined(OF_VIDEO_PLAYER_QUICKTIME)
+//	closeQuicktime();
+//	#endif
 
 
 	//------------------------
diff --git a/libs/openFrameworks/graphics/ofTrueTypeFont.cpp b/libs/openFrameworks/graphics/ofTrueTypeFont.cpp
index b475c9a..9f9bcc7 100644
--- a/libs/openFrameworks/graphics/ofTrueTypeFont.cpp
+++ b/libs/openFrameworks/graphics/ofTrueTypeFont.cpp
@@ -2,10 +2,10 @@
 //--------------------------
 
 #include "ft2build.h"
-#include "freetype2/freetype/freetype.h"
-#include "freetype2/freetype/ftglyph.h"
-#include "freetype2/freetype/ftoutln.h"
-#include "freetype2/freetype/fttrigon.h"
+#include "freetype2/freetype.h"
+#include "freetype2/ftglyph.h"
+#include "freetype2/ftoutln.h"
+#include "freetype2/fttrigon.h"
 
 #ifdef TARGET_LINUX
 #include <fontconfig/fontconfig.h>
diff --git a/libs/openFrameworks/ofMain.h b/libs/openFrameworks/ofMain.h
index aae2394..22e8b9c 100644
--- a/libs/openFrameworks/ofMain.h
+++ b/libs/openFrameworks/ofMain.h
@@ -68,13 +68,13 @@
 
 //--------------------------
 // audio
-#include "ofSoundStream.h"
-#include "ofSoundPlayer.h"
+//#include "ofSoundStream.h"
+//#include "ofSoundPlayer.h"
 
 //--------------------------
 // video
-#include "ofVideoGrabber.h"
-#include "ofVideoPlayer.h"
+//#include "ofVideoGrabber.h"
+//#include "ofVideoPlayer.h"
 
 //--------------------------
 // events
diff --git a/libs/openFrameworks/utils/ofConstants.h b/libs/openFrameworks/utils/ofConstants.h
index 9c86079..1156e5a 100644
--- a/libs/openFrameworks/utils/ofConstants.h
+++ b/libs/openFrameworks/utils/ofConstants.h
@@ -96,8 +96,8 @@ enum ofTargetPlatform{
 
 	#include <windows.h>
 	#define GLEW_STATIC
-	#include "GL\glew.h"
-	#include "GL\wglew.h"
+    #include "GL/glew.h"
+    #include "GL/wglew.h"
    	#include "glu.h"
 	#define __WINDOWS_DS__
 	#define __WINDOWS_MM__
