--- a/openframeworks/app/ofAppGlutWindow.cpp
+++ b/openframeworks/app/ofAppGlutWindow.cpp
@@ -169,10 +169,10 @@ static void fixCloseWindowOnWin32(){
 	DragAcceptFiles (handle, TRUE);
 
 	//store the current message event handler for the window
-	currentWndProc = (WNDPROC)GetWindowLongPtr(handle, GWL_WNDPROC);
+	currentWndProc = (WNDPROC)GetWindowLongPtr(handle, GWLP_WNDPROC);
 
 	//tell the window to now use our event handler!
-	SetWindowLongPtr(handle, GWL_WNDPROC, (long)winProc);
+	SetWindowLongPtr(handle, GWLP_WNDPROC, (long long)winProc);
 }
 
 #endif
--- a/openframeworks/graphics/ofTrueTypeFont.cpp
+++ b/openframeworks/graphics/ofTrueTypeFont.cpp
@@ -10,10 +10,10 @@
 #include FT_TRIGONOMETRY_H
 #include <fontconfig/fontconfig.h>
 #else
-#include "freetype2/freetype/freetype.h"
-#include "freetype2/freetype/ftglyph.h"
-#include "freetype2/freetype/ftoutln.h"
-#include "freetype2/freetype/fttrigon.h"
+#include "freetype2/freetype.h"
+#include "freetype2/ftglyph.h"
+#include "freetype2/ftoutln.h"
+#include "freetype2/fttrigon.h"
 #endif
 
 #include <algorithm>
