--- a/openframeworks/events/ofEvents.cpp
+++ b/openframeworks/events/ofEvents.cpp
@@ -145,29 +145,6 @@
 
 //------------------------------------------
 void ofNotifyKeyPressed(int key, int keycode, int scancode, int codepoint){
-	// FIXME: modifiers are being reported twice, for generic and for left/right
-	// add operators to the arguments class so it can be checked for both
-    if(key == OF_KEY_RIGHT_CONTROL || key == OF_KEY_LEFT_CONTROL){
-        pressedKeys.insert(OF_KEY_CONTROL);
-        ofKeyEventArgs keyEventArgs(ofKeyEventArgs::Pressed,OF_KEY_CONTROL);
-        ofNotifyEvent( ofEvents().keyPressed, keyEventArgs );
-    }
-    else if(key == OF_KEY_RIGHT_SHIFT || key == OF_KEY_LEFT_SHIFT){
-        pressedKeys.insert(OF_KEY_SHIFT);
-        ofKeyEventArgs keyEventArgs(ofKeyEventArgs::Pressed,OF_KEY_SHIFT);
-        ofNotifyEvent( ofEvents().keyPressed, keyEventArgs );
-    }
-    else if(key == OF_KEY_LEFT_ALT || key == OF_KEY_RIGHT_ALT){
-        pressedKeys.insert(OF_KEY_ALT);
-        ofKeyEventArgs keyEventArgs(ofKeyEventArgs::Pressed,OF_KEY_ALT);
-        ofNotifyEvent( ofEvents().keyPressed, keyEventArgs );
-    }
-    else if(key == OF_KEY_LEFT_SUPER || key == OF_KEY_RIGHT_SUPER){
-        pressedKeys.insert(OF_KEY_SUPER);
-        ofKeyEventArgs keyEventArgs(ofKeyEventArgs::Pressed,OF_KEY_SUPER);
-        ofNotifyEvent( ofEvents().keyPressed, keyEventArgs );
-    }
-            
 	pressedKeys.insert(key);
     ofKeyEventArgs keyEventArgs(ofKeyEventArgs::Pressed,key,keycode,scancode,codepoint);
 	ofNotifyEvent( ofEvents().keyPressed, keyEventArgs );
@@ -182,29 +159,6 @@
 
 //------------------------------------------
 void ofNotifyKeyReleased(int key, int keycode, int scancode, int codepoint){
-	// FIXME: modifiers are being reported twice, for generic and for left/right
-	// add operators to the arguments class so it can be checked for both
-    if(key == OF_KEY_RIGHT_CONTROL || key == OF_KEY_LEFT_CONTROL){
-        pressedKeys.erase(OF_KEY_CONTROL);
-        ofKeyEventArgs keyEventArgs(ofKeyEventArgs::Released,OF_KEY_CONTROL);
-    	ofNotifyEvent( ofEvents().keyReleased, keyEventArgs );
-    }
-    else if(key == OF_KEY_RIGHT_SHIFT || key == OF_KEY_LEFT_SHIFT){
-        pressedKeys.erase(OF_KEY_SHIFT);
-        ofKeyEventArgs keyEventArgs(ofKeyEventArgs::Released,OF_KEY_SHIFT);
-    	ofNotifyEvent( ofEvents().keyReleased, keyEventArgs );
-    }
-    else if(key == OF_KEY_LEFT_ALT || key == OF_KEY_RIGHT_ALT){
-        pressedKeys.erase(OF_KEY_ALT);
-        ofKeyEventArgs keyEventArgs(ofKeyEventArgs::Released,OF_KEY_ALT);
-    	ofNotifyEvent( ofEvents().keyReleased, keyEventArgs );
-    }
-    else if(key == OF_KEY_LEFT_SUPER || key == OF_KEY_RIGHT_SUPER){
-        pressedKeys.erase(OF_KEY_SUPER);
-        ofKeyEventArgs keyEventArgs(ofKeyEventArgs::Released,OF_KEY_SUPER);
-    	ofNotifyEvent( ofEvents().keyReleased, keyEventArgs );
-    }
-    
 	pressedKeys.erase(key);
     ofKeyEventArgs keyEventArgs(ofKeyEventArgs::Released,key,keycode,scancode,codepoint);
 	ofNotifyEvent( ofEvents().keyReleased, keyEventArgs );
