--- a/openframeworks/graphics/ofGraphics.cpp
+++ b/openframeworks/graphics/ofGraphics.cpp
@@ -35,17 +35,6 @@
     #define CALLBACK
 #endif
 
-#ifdef TARGET_WIN32
-	#define GLUT_BUILDING_LIB
-	#include "glut.h"
-#endif
-#ifdef TARGET_OSX
-	#include <GLUT/glut.h>
-#endif
-#if defined( TARGET_LINUX ) && !defined(TARGET_OPENGLES)
-	#include <GL/glut.h>
-#endif
-
 
 //style stuff - new in 006
 static ofStyle currentStyle;
