--- a/openframeworks/app/ofAppGLFWWindow.cpp
+++ b/openframeworks/app/ofAppGLFWWindow.cpp
@@ -923,7 +923,8 @@
 
 //------------------------------------------------------------
 void ofAppGLFWWindow::scroll_cb(GLFWwindow* windowP_, double x, double y) {
-	//TODO: implement scroll events
+	int button = y == 1? OF_MOUSE_BUTTON_4 : OF_MOUSE_BUTTON_5;
+	ofNotifyMousePressed(ofGetMouseX(), ofGetMouseY(), button);
 }
 
 //------------------------------------------------------------
