--- a/ofMain.h
+++ b/ofMain.h
@@ -78,9 +78,6 @@
 #include "ofMainLoop.h"
 #if !defined( TARGET_OF_IOS ) & !defined(TARGET_ANDROID) & !defined(TARGET_EMSCRIPTEN)
 	#include "ofAppGLFWWindow.h"
-	#if !defined( TARGET_LINUX_ARM )
-		#include "ofAppGlutWindow.h"
-	#endif
 #endif
 
 //--------------------------
--- a/utils/ofConstants.h
+++ b/utils/ofConstants.h
@@ -98,7 +98,7 @@
 	#define GLEW_STATIC
 	#include "GL/glew.h"
 	#include "GL/wglew.h"
-   	#include "glu.h"
+	#include "GL/glu.h"
 	#define __WINDOWS_DS__
 	#define __WINDOWS_MM__
 	#if (_MSC_VER)       // microsoft visual studio
--- a/utils/ofFileUtils.cpp
+++ b/utils/ofFileUtils.cpp
@@ -565,7 +565,7 @@
 bool ofFile::canRead() const {
 	auto perm = std::filesystem::status(myFile).permissions();
 #ifdef TARGET_WIN32
-	DWORD attr = GetFileAttributes(myFile.native().c_str());
+	DWORD attr = GetFileAttributes((LPSTR)myFile.native().c_str());
 	if (attr == INVALID_FILE_ATTRIBUTES)
 	{
 		return false;
@@ -588,7 +588,7 @@
 bool ofFile::canWrite() const {
 	auto perm = std::filesystem::status(myFile).permissions();
 #ifdef TARGET_WIN32
-	DWORD attr = GetFileAttributes(myFile.native().c_str());
+	DWORD attr = GetFileAttributes((LPSTR)myFile.native().c_str());
 	if (attr == INVALID_FILE_ATTRIBUTES){
 		return false;
 	}else{
