--- a/ofMain.h
+++ b/ofMain.h
@@ -78,9 +78,6 @@
 #include "ofMainLoop.h"
 #if !defined( TARGET_OF_IOS ) & !defined(TARGET_ANDROID) & !defined(TARGET_EMSCRIPTEN)
 	#include "ofAppGLFWWindow.h"
-	#if !defined( TARGET_LINUX_ARM )
-		#include "ofAppGlutWindow.h"
-	#endif
 #endif
 
 //--------------------------
--- a/utils/ofConstants.h
+++ b/utils/ofConstants.h
@@ -98,7 +98,7 @@
 	#define GLEW_STATIC
 	#include "GL/glew.h"
 	#include "GL/wglew.h"
-   	#include "glu.h"
+	#include "GL/glu.h"
 	#define __WINDOWS_DS__
 	#define __WINDOWS_MM__
 	#if (_MSC_VER)       // microsoft visual studio
--- a/sound/ofOpenALSoundPlayer.h
+++ b/sound/ofOpenALSoundPlayer.h
@@ -19,6 +19,7 @@
 #include "kiss_fftr.h"
 #include <sndfile.h>
 #ifdef OF_USING_MPG123
+	#include <unistd.h>
 	#include <mpg123.h>
 #endif
 
--- a/utils/ofConstants.h
+++ b/utils/ofConstants.h
@@ -351,10 +351,10 @@
 // xcode has a bug where it won't support tls on some versions even
 // on c++11, this is a workaround that bug
 #if !defined(TARGET_OSX) && !defined(TARGET_OF_IOS)
- #define HAS_TLS 1
+ #define HAS_TLS 0
 #elif __clang__
  #if __has_feature(cxx_thread_local)
-   #define HAS_TLS 1
+   #define HAS_TLS 0
  #endif
 #endif
 