--- a/openframeworks/ofMain.h
+++ b/openframeworks/ofMain.h
@@ -75,9 +75,6 @@
 #include "ofMainLoop.h"
 #if !defined( TARGET_OF_IOS ) & !defined(TARGET_ANDROID) & !defined(TARGET_EMSCRIPTEN)
 	#include "ofAppGLFWWindow.h"
-	#if !defined( TARGET_LINUX_ARM )
-		#include "ofAppGlutWindow.h"
-	#endif
 #endif
 
 //--------------------------
--- a/openframeworks/app/ofAppRunner.cpp
+++ b/openframeworks/app/ofAppRunner.cpp
@@ -21,7 +21,7 @@
 
 
 // adding this for vc2010 compile: error C3861: 'closeQuicktime': identifier not found
-#if defined (TARGET_WIN32) || defined(TARGET_OSX)
+#if defined (OF_VIDEO_PLAYER_QUICKTIME) || defined(TARGET_OSX)
 	#include "ofQtUtils.h"
 #endif
 
--- a/openframeworks/video/ofDirectShowPlayer.cpp
+++ b/openframeworks/video/ofDirectShowPlayer.cpp
@@ -14,15 +14,14 @@
 //-------------------------------------------------------------------------------------------------------------------------------------------------------------
 
 
-#include <DShow.h>
+#include <dshow.h>
 #pragma include_alias( "dxtrans.h", "qedit.h" )
 #define __IDxtCompositor_INTERFACE_DEFINED__
 #define __IDxtAlphaSetter_INTERFACE_DEFINED__
 #define __IDxtJpeg_INTERFACE_DEFINED__
 #define __IDxtKey_INTERFACE_DEFINED__
-#include <uuids.h>
-#include <Aviriff.h>
-#include <Windows.h>
+#include <aviriff.h>
+#include <windows.h>
 
 //for threading
 #include <process.h>
--- a/openframeworks/app/ofAppGlutWindow.cpp
+++ b/openframeworks/app/ofAppGlutWindow.cpp
@@ -6,7 +6,7 @@
 
 #ifdef TARGET_WIN32
 	#define GLUT_BUILDING_LIB
-	#include "glut.h"
+	#include "GL/glut.h"
 #endif
 #ifdef TARGET_OSX
     #include <OpenGL/OpenGL.h>
--- a/openframeworks/utils/ofConstants.h
+++ b/openframeworks/utils/ofConstants.h
@@ -110,7 +110,7 @@
 	#define GLEW_STATIC
 	#include "GL/glew.h"
 	#include "GL/wglew.h"
-   	#include "glu.h"
+	#include "GL/glu.h"
 	#define __WINDOWS_DS__
 	#define __WINDOWS_MM__
 	#if (_MSC_VER)       // microsoft visual studio
--- a/openframeworks/app/ofAppRunner.cpp
+++ b/openframeworks/app/ofAppRunner.cpp
@@ -73,7 +73,7 @@
 
 #if defined(TARGET_ANDROID) || defined(TARGET_OF_IOS)
     // manage own exit
-#else
+#elif !defined(__MINGW32__)
 	atexit(ofExitCallback);
 #endif
 
