--- a/gl/ofShader.h
+++ b/gl/ofShader.h
@@ -170,9 +170,9 @@
 		std::string sourcePath;
 	};
 
-	unordered_map<GLenum, Shader> shaders;
-	unordered_map<string, GLint> uniformsCache;
-	mutable unordered_map<string, GLint> attributesBindingsCache;
+	map<GLenum, Shader> shaders;
+	map<string, GLint> uniformsCache;
+	mutable map<string, GLint> attributesBindingsCache;
 
 	void checkProgramInfoLog(GLuint program);
 	bool checkProgramLinkStatus(GLuint program);
