--- a/Foundation/src/EventLogChannel.cpp
+++ b/Foundation/src/EventLogChannel.cpp
@@ -17,7 +17,6 @@
 #include "Poco/EventLogChannel.h"
 #include "Poco/Message.h"
 #include "Poco/String.h"
-#include "pocomsg.h"
 #if defined(POCO_WIN32_UTF8)
 #include "Poco/UnicodeConverter.h"
 #endif
--- a/Foundation/include/Poco/Config.h
+++ b/Foundation/include/Poco/Config.h
@@ -23,11 +23,15 @@
 // Define to enable Windows Unicode (UTF-8) support
 // NOTE: As of POCO C++ Libraries release 1.6.0, compiling POCO
 // without POCO_WIN32_UTF8 defined on Windows is deprecated.
+#ifndef POCO_WIN32_UTF8
 #define POCO_WIN32_UTF8
+#endif
 
 
 // Define to enable C++11 support
-// #define POCO_ENABLE_CPP11
+#ifndef POCO_ENABLE_CPP11
+#define POCO_ENABLE_CPP11
+#endif
 
 
 // Define to disable implicit linking
