--- a/poco/NetSSL_Win/src/Context.cpp
+++ b/poco/NetSSL_Win/src/Context.cpp
@@ -218,7 +218,7 @@
 	blob.cbData = static_cast<DWORD>(size);
 	blob.pbData = reinterpret_cast<BYTE*>(const_cast<char*>(pBuffer));
 
-	HCERTSTORE hTempStore =  PFXImportCertStore(&blob, wpassword.data(), PKCS12_ALLOW_OVERWRITE_KEY | PKCS12_INCLUDE_EXTENDED_PROPERTIES);
+	HCERTSTORE hTempStore =  PFXImportCertStore(&blob, wpassword.data(), 0x00004000/*PKCS12_ALLOW_OVERWRITE_KEY*/ | 0x0010/*PKCS12_INCLUDE_EXTENDED_PROPERTIES*/);
 	
 	// clear UTF-16 password
 	std::fill(const_cast<wchar_t*>(wpassword.data()), const_cast<wchar_t*>(wpassword.data() + password.size()), L'X');
@@ -272,7 +272,7 @@
 	if (_pCert)
 	{
 		schannelCred.cCreds = 1; // how many cred are stored in &pCertContext
-		schannelCred.paCred = &const_cast<PCCERT_CONTEXT>(_pCert);
+		schannelCred.paCred = const_cast<PCCERT_CONTEXT*>(&_pCert);
 	}
 
 	schannelCred.grbitEnabledProtocols = proto();
--- a/poco/Foundation/src/EventLogChannel.cpp
+++ b/poco/Foundation/src/EventLogChannel.cpp
@@ -17,7 +17,6 @@
 #include "Poco/EventLogChannel.h"
 #include "Poco/Message.h"
 #include "Poco/String.h"
-#include "pocomsg.h"
 #if defined(POCO_WIN32_UTF8)
 #include "Poco/UnicodeConverter.h"
 #endif
--- a/poco/Foundation/include/Poco/Config.h
+++ b/poco/Foundation/include/Poco/Config.h
@@ -23,11 +23,15 @@
 // Define to enable Windows Unicode (UTF-8) support
 // NOTE: As of POCO C++ Libraries release 1.6.0, compiling POCO
 // without POCO_WIN32_UTF8 defined on Windows is deprecated.
+#ifndef POCO_WIN32_UTF8
 #define POCO_WIN32_UTF8
+#endif
 
 
 // Define to enable C++11 support
-// #define POCO_ENABLE_CPP11
+#ifndef POCO_ENABLE_CPP11
+#define POCO_ENABLE_CPP11
+#endif
 
 
 // Define to disable implicit linking
@@ -39,7 +43,7 @@
 // initialization framework-wide (e.g. Net
 // on Windows, all Data back-ends, etc).
 // 
-// #define POCO_NO_AUTOMATIC_LIB_INIT
+#define POCO_NO_AUTOMATIC_LIB_INIT
 
 
 // Define to disable FPEnvironment support
@@ -90,7 +94,7 @@
 // !!! for std::aligned_storage.                           !!!
 // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 // 
-#define POCO_NO_SOO
+// #define POCO_NO_SOO
 
 
 // Small object size in bytes. When assigned to Any or Var,
